### PR TITLE
organize platforms specific files in sub-directories

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

**Review by m-kuhn — APPROVED** [review]

File: src/core/qgismobileapp.h [hunk 1]
```diff
@@ -40,7 +40,7 @@
 #if defined( Q_OS_ANDROID )
```
Comment (m-kuhn) [inline src/core/qgismobileapp.h:40]: I think these platform specific imports should no(t) (longer) be required

**Comment by nirvn** [comment]
@3nids , while your having fun moving files around, there are two utils classes outside of the utils directory :) expression context utils and snapping utils.

File: src/core/qgismobileapp.h (hunk 1 above)
Comment (3nids) [reply src/core/qgismobileapp.h]: done let's see

**Comment by 3nids** [comment]
@nirvn I'll do that in a second step

**Comment by m-kuhn** [comment]
@3nids do you think it'd be possible to kill the ifdefs in `qgismobileapp.cpp`?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 12f8d19958d0223595652f895c151b9ff95da2af:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1988-d57b4d12d0bf310ea92d910a6695d49d2ccc9426-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1988-d57b4d12d0bf310ea92d910a6695d49d2ccc9426-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1988-d57b4d12d0bf310ea92d910a6695d49d2ccc9426-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1988-d57b4d12d0bf310ea92d910a6695d49d2ccc9426-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 2fa0134e9d64bdd4f8f2d7db14da8fb0cf0a2ba2:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1988-fe4509cb7934104eca986d855f0ff53013307329-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1988-fe4509cb7934104eca986d855f0ff53013307329-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1988-fe4509cb7934104eca986d855f0ff53013307329-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1988-fe4509cb7934104eca986d855f0ff53013307329-x86.apk)

**Pull request merged**